modules/auxiliary/scanner/ssh/ssh_version.rb

```diff
@@ -4,96 +4,275 @@
+    }
+
+    server_data[:encryption_server].each do |encryption|
+      note = ''
+      encryption_checks.each do |bad_enc, refs|
+        next unless encryption.downcase == bad_enc
 
-      # Check to see if this is Kippo, which sends a premature
-      # key init exchange right on top of the SSH version without
-      # waiting for the required client identification string.
-      if first_message && first_message.size >= 5
-        extra = first_message.unpack("NCCA*") # sz, pad_sz, code, data
-        if (extra.last.size + 2 == extra[0]) && extra[2] == 20
-          info << " (Kippo Honeypot)"
+        vprint_good("#{target_host} - Encryption #{encryption} is deprecated and should not be used.")
+        report_vuln(
+          host: target_host,
+          port: rport,
+          proto: 'tcp',
+          name: name,
+          info: "Module #{fullname} confirmed SSH Encryption #{encryption} is available, but should be deprecated",
+          refs: refs
+        )
+        note = 'Deprecated'
+      end
+      table << ['encryption.encryption', encryption, note]
+    end
+    table
+  end
+
+  def check_kex(server_data)
+    table = []
+    kex_checks = {
+      'gss-group1-sha1-*' => ['https://datatracker.ietf.org/doc/html/rfc8732#name-deprecated-algorithms'],
+      'gss-group14-sha1-gss-gex-sha1-*' => ['https://datatracker.ietf.org/doc/html/rfc8732#name-deprecated-algorithms'],
+      'gss-gex-sha1-*' => ['https://datatracker.ietf.org/doc/html/rfc8732#name-deprecated-algorithms'],
+      'ecdsa-sha2-nistp521' => ['https://github.com/net-ssh/net-ssh?tab=readme-ov-file#key-exchange'],
+      'ecdsa-sha2-nistp384' => ['https://github.com/net-ssh/net-ssh?tab=readme-ov-file#key-exchange'],
+      'ecdsa-sha2-nistp256' => ['https://github.com/net-ssh/net-ssh?tab=readme-ov-file#key-exchange'],
+      'diffie-hellman-group-exchange-sha1' => ['https://datatracker.ietf.org/doc/html/draft-ietf-curdle-ssh-kex-sha2-20#page-16'],
+      'diffie-hellman-group1-sha1' => ['https://datatracker.ietf.org/doc/html/draft-ietf-curdle-ssh-kex-sha2-20#page-16'],
+      'rsa1024-sha1' => ['https://datatracker.ietf.org/doc/html/draft-ietf-curdle-ssh-kex-sha2-20#page-16']
+    }
+    server_data[:kex].each do |kex|
+      note = ''
+      kex_checks.each do |bad_kex, refs|
+        if bad_kex.ends_with? '*'
+          next unless kex.downcase.start_with? bad_kex[0..-2]
+        else
+          next unless kex.downcase == bad_kex
         end
+
+        vprint_good("#{target_host} - Key Exchange (kex) #{kex} is deprecated and should not be used.")
+        report_vuln(
+          host: target_host,
+          port: rport,
+          proto: 'tcp',
+          name: name,
+          info: "Module #{fullname} confirmed SSH Encryption #{kex} is available, but should be deprecated",
+          refs: refs
+        )
+        note = 'Deprecated'
+      end
+      table << ['encryption.key_exchange_(kex)', kex, note]
```

Comment:
Shouldn't this just be `encryption.key_exchange` since kex is short for key exchange?

Reply:
yes, I added the (kex) since not everyone may be familiar with the term, but easy to remove.